File: game-logic-core/scala3MainProject/src/main/scala/fwc/game/actionPhase/CombatOutcome.scala
package fwc.game.actionPhase

import fwc.{JsonParsable, JsonSerializable}
import fwc.game.houses.HouseType
import ujson.Value

case class CombatOutcome(
                          attackerStrength: Int,
                          defenderStrength: Int,
                          winner: Option[HouseType],
                          attackerUnitsToKill: Int,
                          defenderUnitsToKill: Int,
                        ) extends JsonSerializable {
  override def toJson: Value = ujson.Obj(
    "attackerStrength" -> attackerStrength,
    "defenderStrength" -> defenderStrength,
    "winner" -> (if winner.isEmpty then ujson.Null else winner.head.toString),
    "attackerUnitsToKill" -> attackerUnitsToKill,
    "defenderUnitsToKill" -> defenderUnitsToKill
  )
}

object CombatOutcome extends JsonParsable {
  override def fromJson(json: Value): CombatOutcome = {
    CombatOutcome(
      json("attackerStrength").num.toInt,
      json("defenderStrength").num.toInt,
      if json("winner") == ujson.Null then None else Some(HouseType.fromString(json("winner").str)),
      json("attackerUnitsToKill").num.toInt,
      json("defenderUnitsToKill").num.toInt,
    )
  }
}

File: bots/server_module/game_state/combat_outcome.py
from server_module.game_state.house_type import HouseType


class CombatOutcome:
    def __init__(
            self,
            attacker_strength: int,
            defender_strength: int,
            attacker_units_to_kill: int,
            defender_units_to_kill: int,
            winner: HouseType,
    ):
        self.attacker_strength = attacker_strength
        self.defender_strength = defender_strength
        self.attacker_units_to_kill = attacker_units_to_kill
        self.defender_units_to_kill = defender_units_to_kill
        self.winner = winner

